### ISSUE TITLE
Trouble with referencing server side endpoint from view attribute in compose element

### ISSUE THREAD

**Issue by deap82** [dumberjs/aurelia-deps-finder#1]
In my aurelia app, which lives inside an asp.net mvc site I have a situation where I need some markup to be rendered server side so I have a compose element like this;

`<compose view="/UserComments/QuickCommentForm"></compose>`

"/UserComments/QuickCommentForm" is the route for a server side endpoint returning the html needed.

When replacing jspm with dumber my bundling task gives an error saying it can't find any package named "UserComments".

![image](https://user-images.githubusercontent.com/1295067/164149201-52d43a20-7038-44c7-8a95-c0a663d95ef0.png)

I changed it to 

`<compose view.bind="'/UserComments/QuickCommentForm'"></compose>`

and the bundling task is successful.

But when running the app, " /UserComments/QuickCommentForm._**js**_" is requested. So I added

`[Route("UserComments/QuickCommentForm.js")]`

to the action in the MVC controller and now it works, **_however, this feels a bit hackish..._**

ALSO, this component is used multiple times on the same page and before switching to dumber the server side endpoint was only called once and the result is then reused for all instances. I have always thought that aurelia does some kind of caching of views and was grateful for the behavior. Running the app now (bundled with dumber) and going to the page in question, inspecting the networks tab of browser developer tools I can see that it is requested multiple times, ones for each instance;

![image](https://user-images.githubusercontent.com/1295067/164150362-e0b4cd25-4a7a-47df-b465-750ca8671cc8.png)

I also found this in the dumber documentation;

![image](https://user-images.githubusercontent.com/1295067/164150424-78803d5b-4565-4cd8-8618-bac01ea848fb.png)

How can I get this path to be regarded as a "local source file" so that it's just ignored and can I get back to the old behavior where the endpoint is only called once? (For the last part I'm thinking it might be related to me changing from "view" to "view.bind" which was only done to hide it from aurelia-deps-finder.)

Any input greatly appreciated.

**Comment by 3cp** [comment]
The first issue is probably mine. It should not try npm package when the module requested is starting with "/". I will get back to you.

The second one is the nature of requirejs, it tries to load module `/UserComments/QuickCommentForm`, but doesn't know it's a html view (means should be plain text. So it added ".js" assuming the module should come from a JavaScript file.

You can try to use route `/UserComments/QuickCommentForm.html`, the ".html" extname will go through some default setting in dumber-module-loader: loading html file as plain text.

**Comment by deap82** [comment]
@3cp Using .html and removing .bind;

`<compose view="/UserComments/QuickCommentForm.html"></compose>`

and this on the server side endpoint;

`[Route("UserComments/QuickCommentForm.html")]`

both bundling it and running it now works.

But endpoint is still requested multiple times... I'm not really sure if the previous behavior with one request to the server for multiple instances of the frontend component is aurelia or systemjs-related... But since what I'm doing right now is only about replacing jspm/systemjs with dumber it seems to be the latter...

**Comment by 3cp** [comment]
Please clarify:
`<compose view="/UserComments/QuickCommentForm.html"></compose>` does not fail "cannot find npm package" anymore?

"be the latter..." I guess you mean the former? aurelia.

The multiple requests to same html file is unusual. I will check that.

**Comment by 3cp** [comment]
I just tested it. It only issues one remote request in my app.

If you have CORS setup, means frontend and backend are on two different hostnames, you will see two requests, first one is OPTIONS, second one is GET.

**Comment by 3cp** [comment]
Finally fixed the "cannot find npm package" error.
The patch is in aurelia-deps-finder and also dumber-module-loader.

After reinstall deps, you need to do `npm run clear-cache` to reset dumber's cache (the cache is associated with version of dumber which is not the npm package patched for this issue).

You can go back to use the route without `.html`.
`<compose view="/UserComments/QuickCommentForm"></compose>` should work.

**Comment by 3cp** [comment]
Let me know if it all worked out. Thx.

**Comment by deap82** [comment]
@3cp Sorry about the delay, other stuff got in the way but I got around to check on this again today.

It now works to remove the ".html" ext hack and I'm not getting any bundling error about missing npm package :+1:. 

However I still have the change in server side request behavior. Before changing to use dumber.js I'm only getting one request to "/UserComments/QuickCommentForm" when loading a view that has several instances of the custom element. With dumber.js I'm getting one request per instance.

**What I noticed now though** is that leaving that view and then returning to it no repeated requests are made. Also, the view has an infinite scroll behavior loading more posts as I scroll the page, those requests also does not trigger the "/UserComments/QuickCommentForm" endpoint to be requested again.

So it just seems as if aurelias awareness of the view already been loaded is somehow off on the initial load. Sort of as if something now is going faster than before, so when loading multiple instances of the element "in the same run", all of them runs the request before it has been recorded that it's already been fetched. When additonal instances are rendered at a later point in time (i.e. when scrolling the page or leaving and returning to the page) aurelia has recorded the html view as loaded and no new requests are made. (This is just a theoretic reasoning of mine, I would agree that just changing the bundler shouldn't be able to cause this kind of change...)

**So, a kind of workaround I can do now** is to include the custom element in a hidden container within my layout markup. That way, no additional requests are made when accessing the page with several instances. However, reloading the browser when standing on that page then still causes multiple requests.

I should point out that a container hosting these custom elements are "aureliaized" with the aurelia enhance api, if that might be related somehow.
